test/parallel/test-pipe-outgoing-message-data-emitted-after-ended.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+const common = require('../common');
+const http = require('http');
+const OutgoingMessage = require('_http_outgoing').OutgoingMessage;
+
+const server = http.createServer(common.mustCall(function(req, res) {
+  console.log('Got a request, piping an OutgoingMessage to it.');
+  const outgointMessage = new OutgoingMessage();
+  outgointMessage.pipe(res);
```

Comment:
Again, I agree. But that's how request npm works... 
I guess this should either be disabled (by throwing an exception), or it should work.
Here is what they do in request npm:
1) They call to `pipe` with the `OutgoingMessage`:
https://github.com/request/request/blob/master/request.js#L1489
2) Upon data on the `IncomingResponse`, they emit this on the `OutgoingMessage`:
https://github.com/request/request/blob/master/request.js#L1082

That's **really** an odd implementation of `pipe`, I agree. But since it is not throwing an error, I think it should work (or being disabled in the code).